internal/backup/volumesnapshot_action.go

```diff
@@ -60,8 +60,24 @@ func (p *VolumeSnapshotBackupItemAction) Execute(item runtime.Unstructured, back
 		return nil, nil, errors.WithStack(err)
 	}
 
+	// determine if we are backing up a volumesnapshot that was created by velero while performing backup of a
+	// CSI backed PVC.
+	// For volumesnapshots that were created during the backup of a CSI backed PVC, we will wait for the volumecontents to
+	// be available.
+	// For volumesnapshots created outside of velero, we expect the volumesnapshotcontent to be available prior to backing up
+	// the volumesnapshot. In case of a failure, backup should be re-attempted after the CSI driver has reconciled the volumesnapshot.
+	// existence of the velerov1api.BackupNameLabel indicates that the volumesnapshot was created while backing up a
+	// CSI backed PVC.
+
+	// TODO: use UID of the backup object instead of just looking for a match in the backup name. This will ensure we don't wait in cases where
+	// backup names are reused.
```

Comment:
Once we have proper garbage collection, I don't think the re-use case should matter, should it?

When deleting the `Backup`, all namespaced `VolumeSnapshot`s should also be deleted, though any `VolumeSnapshotContents` objects with a deletion policy of `Retain` won't be able to be touched.

So we can't re-use the `Backup` name while the object exists, and deleting the `Backup` manually via `kubectl` is unsupported, requiring manual clean up....or we have our GC look at `VolumeSnapshots` pointing to a `Backup` that doesn't exist and delete them for the user.

Reply:
Yes, once we have proper garbage collection. This shouldn't matter.
We can remove the UID `//TODO` when we've implemented the backup deletion.